lib/jekyll/frontmatter_defaults.rb

```diff
@@ -123,6 +128,13 @@ def path_is_subpath?(path, parent_path)
       false
     end
 
+    def glob_pattern?(path)
+      path.each_filename do |str|
+        return true if str =~ %r!\*+|\?|\[.*\]|{.*}!
```

Comment:
Does `path.include?(“*”)` not work for this case?

Reply:
I wanted to test the path component string for specific glob patterns. and regex seems better suited for it than multiple `String#include?` calls

Reply:
@ashmaroli Why do we need to test each filename separately? AFAIK, the presence of `*` in `scope["path"]` is enough to know it's a glob pattern.

Reply:
@parkr 
```
defaults:
  - scope:
      path: "_puppies/arrears.?"
      type: "puppies"
    values:
      layout: arrears
``` 
is also a glob pattern to catch all kinds of files with basename `arrears`, if you ask me..